### ISSUE TITLE
PLSuperExpCutoff2 as spectral form

### ISSUE THREAD

**Issue by zimmerst** [fermiPy/fermipy#200]
In FL8Y.fit the new spectral form PLSuperExpCutoff2 is introduced. It doesn't have an equivalent mapping in fermipy:

````
['ConstantValue',
 'LogParabola',
 'EblAtten::SmoothBrokenPowerLaw',
 'EnergyBand::BandFunction',
 'EblAtten::PLSuperExpCutoff',
 'EnergyBand::SmoothBrokenPowerLaw',
 'RadialGaussian',
 'EblAtten::ExpCutoff',
 'BrokenPowerLaw',
 'EblAtten::BrokenPowerLaw2',
 'DMFitFunction',
 'EblAtten::LogParabola',
 'EblAtten::BandFunction',
 'ExpCutoff',
 'EnergyBand::PLSuperExpCutoff',
 'PLSuperExpCutoff',
 'EnergyBand::PowerLaw2',
 'EnergyBand::LogParabola',
 'CompositeSource',
 'MapCubeFunction',
 'EblAtten::FileFunction',
 'LogGaussian',
 'AbsEdge',
 'LogNormal',
 'SmoothBrokenPowerLaw',
 'ScaleFactor::PowerLaw2',
 'SkyDirFunction',
 'ScaleFactor::FileFunction',
 'LogNormalLog',
 'PowerLaw',
 'SmoothDoubleBrokenPowerLaw',
 'MultipleBrokenPowerLaw',
 'BrokenPowerLaw3',
 'BrokenPowerLaw2',
 'RadialDisk',
 'EblAtten::BPLExpCutoff',
 'EnergyBand::ExpCutoff',
 'BandFunction',
 'ScaleFactor::Gaussian',
 'PowerLaw2',
 'Gaussian',
 'EnergyBand::BrokenPowerLaw2',
 'EnergyBand::BPLExpCutoff',
 'FileFunction',
 'ScaleFactor::PLSuperExpCutoff',
 'EnergyBand::FileFunction',
 'SpatialMap',
 'RadialProfile',
 'PiecewisePowerLaw',
 'ExpCutoffSEDPeak',
 'BPLExpCutoff',
 'EblAtten::PowerLaw2']
````

**Comment by me-manu** [comment]
Which version of fermipy are you using? It's in the latest developer version.

**Comment by dimauromattia** [comment]
Hi @zimmerst I confirm that in version 0.16.0+164.g1712 the new shape PLSuperExpCutoff2 has been included.

**Comment by dimauromattia** [comment]
Hi Stephan, 

I investigated more PLSuperExpCutoff2 and FL8Y list and I am finding the following error when I write an roi and then I put in the catalog entry of the config.yaml, the fits file created by the write_roi command.
I will work on it.

Best Mattia

--------------------------------------------------------------------------------
fermipy version 0.16.0+164.g1712 
ScienceTools version ScienceTools-11-07-00
WARNING: hdu= was not specified but multiple tables are present, reading in first available table (hdu=1) [astropy.io.fits.connect]
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-1-cbe49ea25c0d> in <module>()
     20 args = parser.parse_args()
     21 
---> 22 gta = GTAnalysis(args.config)
     23 #gta = GTAnalysis.create(args.input,args.config) #if you take input files from a yaml file.
     24 

/nfs/slac/kipac/fs1/u/mdimauro/software/fermipy/fermipy/gtanalysis.pyc in __init__(self, config, roi, **kwargs)
    333                                         configs[0]['model'],
    334                                         fileio=configs[0]['fileio'],
--> 335                                         coordsys=configs[0]['binning']['coordsys'])
    336         else:
    337             self._roi = roi

/nfs/slac/kipac/fs1/u/mdimauro/software/fermipy/fermipy/roi_model.pyc in create(cls, selection, config, **kwargs)
   1690         else:
   1691             target_skydir = wcs_utils.get_target_skydir(selection)
-> 1692             return cls.create_from_position(target_skydir, config, **kwargs)
   1693 
   1694     @classmethod

/nfs/slac/kipac/fs1/u/mdimauro/software/fermipy/fermipy/roi_model.pyc in create_from_position(cls, skydir, config, **kwargs)
   1707 
   1708         coordsys = kwargs.pop('coordsys', 'CEL')
-> 1709         roi = cls(config, skydir=skydir, coordsys=coordsys, **kwargs)
   1710         return roi
   1711 

/nfs/slac/kipac/fs1/u/mdimauro/software/fermipy/fermipy/roi_model.pyc in __init__(self, config, **kwargs)
   1318         self._src_radius = []
   1319 
-> 1320         self.load(coordsys=coordsys, srcname=srcname)
   1321 
   1322     def __contains__(self, key):

/nfs/slac/kipac/fs1/u/mdimauro/software/fermipy/fermipy/roi_model.pyc in load(self, **kwargs)
   1642             if extname != '.xml':
   1643                 self.load_fits_catalog(c, extdir=extdir, coordsys=coordsys,
-> 1644                                        srcname=srcname)
   1645             elif extname == '.xml':
   1646                 self.load_xml(c, extdir=extdir, coordsys=coordsys)

/nfs/slac/kipac/fs1/u/mdimauro/software/fermipy/fermipy/roi_model.pyc in load_fits_catalog(self, name, **kwargs)
   1917         # EAC split this function to make it easier to load an existing catalog
   1918         cat = catalog.Catalog.create(name)
-> 1919         self.load_existing_catalog(cat, **kwargs)
   1920 
   1921     def load_existing_catalog(self, cat, **kwargs):

/nfs/slac/kipac/fs1/u/mdimauro/software/fermipy/fermipy/roi_model.pyc in load_existing_catalog(self, cat, **kwargs)
   1981 
   1982             src_dict['spectral_pars'] = spectral_pars_from_catalog(
-> 1983                 catalog_dict)
   1984             src_dict['spatial_pars'] = spatial_pars_from_catalog(catalog_dict)
   1985 

/nfs/slac/kipac/fs1/u/mdimauro/software/fermipy/fermipy/roi_model.pyc in spectral_pars_from_catalog(cat)
    283     elif spectrum_type == 'PLSuperExpCutoff2':
    284 
--> 285         pars['Prefactor']['value'] = cat['params'][0]
    286         pars['Index1']['value'] = cat['params'][1]
    287         pars['Index1']['scale'] = -1.0

KeyError: 'params'

**Comment by dimauromattia** [comment]
I just checked that actually I have the same error also if I work with the 3FGL. With the error for:

/nfs/slac/kipac/fs1/u/mdimauro/software/fermipy/fermipy/roi_model.pyc in spectral_pars_from_catalog(cat)
    267     elif spectrum_type == 'PLSuperExpCutoff':
    268 
--> 269         pars['Prefactor']['value'] = cat['params'][0]
    270         pars['Index1']['value'] = cat['params'][1]
    271         pars['Index1']['scale'] = -1.0

KeyError: 'params'

**Comment by woodmd** [comment]
The issue that @dimauromattia found is now resolved by #206.
